docs/flamingock-library-config/setup-and-stages.md

```diff
@@ -318,3 +322,68 @@ By default, Flamingock automatically scans the following source and resource roo
 - For template-based changes, check file names and YAML formatting
 
 
+## Setup validation
+
+Flamingock validates that all code-based changes (classes annotated with `@Change`) are properly mapped to a stage during compilation.
+
+### Strict stage mapping
+
+By default, Flamingock enforces strict stage mapping validation:
+
+```java
+@EnableFlamingock(
+    stages = { @Stage(location = "com.yourcompany.changes") },
+    strictStageMapping = true  // Default behavior
+)
+public class FlamingockConfig {
+    // Configuration class
+}
+```
+
+When `strictStageMapping` is enabled (default):
+- **Compilation fails** if any `@Change` class is found outside the configured stage locations
+- A `RuntimeException` is thrown at compilation time for unmapped changes
+- Ensures all changes are properly organized and will be executed
+
+When `strictStageMapping` is disabled:
+- **Only warnings are emitted** for unmapped changes during compilation
+- Compilation continues successfully
+- Unmapped changes will be ignored at runtime
+
+### Example scenarios
+
+**Scenario 1: Change outside configured location (strict mode)**
+```java
+@EnableFlamingock(
+    stages = { @Stage(location = "com.yourcompany.changes") },
+    strictStageMapping = true  // Default
+)
+```
+
+If you have a change at `com.yourcompany.yourpackage.OldChange` (outside the configured location):
+- ❌ **Compilation fails** with detailed error message
+- Must move the change to the correct location or add a new stage
+
+**Scenario 2: Relaxed validation**
+```java
+@EnableFlamingock(
+    stages = { @Stage(location = "com.yourcompany.changes") },
+    strictStageMapping = false  // Relaxed mode
+)
+```
+
+If you have a change at `com.yourcompany.yourpackage.OldChange`:
+- ⚠️ **Warning emitted** during compilation
+- Compilation succeeds but change is ignored at runtime
+
+:::tip Best Practice
+Keep `strictStageMapping = true` (default) to ensure all changes are properly mapped and executed. Only disable it temporarily during large refactoring or migration scenarios.
+:::
+
+### Compilation fails with "unmapped changes" error
+- Check that all `@Change` classes are located within the configured stage locations
+- If you have changes in multiple locations, add additional stages to cover them
```

Comment:
"If you have changes in multiple locations, add additional stages to cover them" is tricky, because having multiple stages have some important implications.

I would rewrite it to something like
- If you have changes in multiple locations, **consider** adding an additional stages to cover the them. Visit our multiple-stages sections for more information.(something like this)